controlplane/kubeadm/controllers/kubeadm_control_plane_controller.go

```diff
@@ -339,17 +339,14 @@ func (r *KubeadmControlPlaneReconciler) initializeControlPlane(ctx context.Conte
 func (r *KubeadmControlPlaneReconciler) cloneConfigsAndGenerateMachine(ctx context.Context, cluster *clusterv1.Cluster, kcp *controlplanev1.KubeadmControlPlane, bootstrapSpec *bootstrapv1.KubeadmConfigSpec) error {
 	var errs []error
 
-	ownerRef := metav1.NewControllerRef(kcp, controlplanev1.GroupVersion.WithKind("KubeadmControlPlane"))
-
 	// Clone the infrastructure template
-	infraRef, err := external.CloneTemplate(
-		ctx,
-		r.Client,
-		&kcp.Spec.InfrastructureTemplate,
-		kcp.Namespace,
-		cluster.Name,
-		ownerRef,
-	)
+	infraRef, err := external.CloneTemplate(ctx, &external.CloneTemplateInput{
+		Client:      r.Client,
+		TemplateRef: &kcp.Spec.InfrastructureTemplate,
+		Namespace:   kcp.Namespace,
+		OwnerRef:    metav1.NewControllerRef(kcp, controlplanev1.GroupVersion.WithKind("KubeadmControlPlane")),
+		ClusterName: cluster.Name,
```

Comment:
Should this also apply the labels that we apply to the Machine resource as well (https://github.com/kubernetes-sigs/cluster-api/pull/2091/files#diff-f6190b16a8fe960dc3040118ef7fa34aR432)?

Reply:
We can! I'll send a patch out